app/services/resource_service.py

```diff
@@ -438,3 +448,18 @@ async def fetch_professional_profile_analytics(request_body: object, params: Opt
     # TODO do error-handling over here
 
     return response
+
+
+@retry(attempts=3, delay=1, retry_exceptions=(TooManyRequestException,))
+async def request_async_profile_analytics(request_body: object, params: Optional[Dict]) -> Dict:
```

Comment:
I suppose we are posing a job, If yes let's rename it accordingly.
```suggestion
async def post_async_profile_analytics_request(request_body: object, params: Optional[Dict]) -> Dict:
```